contracts/src/bridge/SequencerInbox.sol

```diff
@@ -176,19 +173,18 @@ contract SequencerInbox is DelegateCallAware, GasRefundEnabled, ISequencerInbox
         IGasRefunder gasRefunder
     ) external override refundsGasNoCalldata(gasRefunder, payable(msg.sender)) {
         if (!isBatchPoster[msg.sender] && msg.sender != rollup) revert NotBatchPoster();
-        if (inboxAccs.length != sequenceNumber) revert BadSequencerNumber();
 
         (bytes32 dataHash, TimeBounds memory timeBounds) = formDataHash(
             data,
             afterDelayedMessagesRead
         );
-        (bytes32 beforeAcc, bytes32 delayedAcc, bytes32 afterAcc) = addSequencerL2BatchImpl(
-            dataHash,
-            afterDelayedMessagesRead
-        );
-        if (data.length > 0) {
-            _reportBatchSpending(dataHash, sequenceNumber);
-        }
+        (
+            uint256 seqMessageCount,
+            bytes32 beforeAcc,
+            bytes32 delayedAcc,
+            bytes32 afterAcc
+        ) = addSequencerL2BatchImpl(dataHash, afterDelayedMessagesRead, data.length);
```

Comment:
For the non-origin variant we shouldn't report batch spending (as the caller may not have paid tx data costs for the calldata). I think we could do that by replacing data.length here with 0.